mirdata/datasets/saraga_hindustani.py

```diff
@@ -0,0 +1,497 @@
+
+    @utils.cached_property
+    def tempo(self):
+        """Dict: tempo annotations"""
+        return load_tempo(self.tempo_path)
+
+    @utils.cached_property
+    def sama(self):
+        """SectionData: sama section annotations"""
+        return load_sama(self.sama_path)
+
+    @utils.cached_property
+    def sections(self):
+        """SectionData: track section annotations"""
+        return load_sections(self.sections_path)
+
+    @utils.cached_property
+    def phrases(self):
+        """EventData: phrase annotations"""
+        return load_phrases(self.phrases_path)
+
+    @property
+    def audio(self):
+        """(np.ndarray, float): audio signal, sample rate"""
+        return load_audio(self.audio_path)
+
+    def to_jams(self):
+        """Jams: the track's data in jams format"""
+        return jams_utils.jams_converter(
+            audio_path=self.audio_path,
+            f0_data=[(self.pitch, 'pitch')],
+            section_data=[(self.sama, 'sama'), (self.sections, 'sections')],
+            event_data=[(self.phrases, 'phrases')],
+            metadata={
+                'tempo': self.tempo,
+                'tonic': self.tonic,
+                'metadata': self._track_metadata,
+            },
+        )
+
+
+def load_audio(audio_path):
+    """Load a Saraga Hindustani audio file.
+
+    Args:
+        audio_path (str): path to audio file
+
+    Returns:
+        y (np.ndarray): the mono audio signal
+        sr (float): The sample rate of the audio file
+
+    """
+    if audio_path is None:
+        return None
+
+    if not os.path.exists(audio_path):
+        raise IOError("audio_path {} does not exist".format(audio_path))
+    return librosa.load(audio_path, sr=44100, mono=False)
+
+
+def load_tonic(tonic_path):
+    """Load track absolute tonic
+
+    Args:
+        tonic_path (str): Local path where the tonic path is stored.
+            If `None`, returns None.
+
+    Returns:
+        (int): Tonic annotation in Hz
+    """
+    if tonic_path is None:
+        return None
+
+    if not os.path.exists(tonic_path):
+        raise IOError("tonic_path {} does not exist".format(tonic_path))
+
+    with open(tonic_path, 'r') as reader:
+        return float(reader.readline().split('\n')[0])
+
+
+def load_pitch(pitch_path):
+    """Load automatic extracted pitch or melody
+
+    Args:
+        pitch path (str): Local path where the pitch annotation is stored.
+            If `None`, returns None.
+
+    Returns:
+        F0Data: pitch annotation
+    """
+    if pitch_path is None:
+        return None
+
+    if not os.path.exists(pitch_path):
+        raise IOError("pitch_path {} does not exist".format(pitch_path))
+
+    times = []
+    freqs = []
+    with open(pitch_path, 'r') as reader:
+        for line in reader.readlines():
+            times.append(float(line.split('\t')[0]))
+            freqs.append(float(line.split('\t')[1]))
+
+    if not times:
+        return None
```

Comment:
I think this means that you have to write a test where times=False so this line is triggered